src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -0,0 +1,240 @@
+
+            GCHandle gch = GCHandle.FromIntPtr(context);
+            var state = (RequestCompletionSource)gch.Target;
+
+            try
+            {
+                CheckDisposed();
+                state.CancellationToken.ThrowIfCancellationRequested();
+
+                Stream contentStream = state.RequestContentStream;
+                if (!contentStream.CanSeek)
+                {
+                    return curlioerr.CURLIOE_FAILRESTART;
+                }
+                contentStream.Seek(0, SeekOrigin.Begin);
+                return curlioerr.CURLIOE_OK;
+            }
+            catch (Exception ex)
+            {
+                state.SavedException = ex;
+#if DEBUG
+                if (Logging.On) Logging.Exception(Logging.Http, this, "CurlSendIoCtlCallback exception for " + Logging.GetObjectLogHash(state.ResponseMessage.ContentStream) + ": ", ex);
+#endif
+                return curlioerr.CURLIOE_FAILRESTART; // Force curl_easy_perform to fail
+            }
+        }
+
+        private ulong ExecuteCallback(
+            IntPtr buffer,
+            ulong size,
+            ulong nitems,
+            IntPtr context,
+            Func<IntPtr, ulong, RequestCompletionSource, ulong> callback)
+        {
+            size *= nitems;
+            GCHandle gch = GCHandle.FromIntPtr(context);
+            var state = (RequestCompletionSource)gch.Target;
+
+            try
+            {
+                CheckDisposed();
+#if DEBUG
+                if (Logging.On)
+                {
+                    Logging.PrintInfo(Logging.Http, this, "Executecallback for " + Logging.GetObjectLogHash(state.ResponseMessage.ContentStream) + " with " + ((size > 0) ? (size + " bytes") : " no data"));
+                }
+#endif
+                return (size == 0) ? 0 : callback(buffer, size, state);
+            }
+            catch (Exception ex)
+            {
+                state.SavedException = ex;
+#if DEBUG
+                if (Logging.On) Logging.Exception(Logging.Http, this, "Executecallback exception for " + Logging.GetObjectLogHash(state.ResponseMessage.ContentStream) + ": ", ex);
+#endif
+                return (size > 0) ? size - 1 : 1; // Force curl_easy_perform to fail
+            }
+        }
+
+        private bool TryParseStatusLine(HttpResponseMessage response, string responseHeader)
+        {
+
+            if (!responseHeader.StartsWith("HTTP/", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+            var tokens = responseHeader.Split(new char[] { ' ' });
+            if (tokens.Length > 1)
+            {
+                response.StatusCode = (HttpStatusCode)int.Parse(tokens[1]);
+                if (tokens.Length > 2)
+                {
+                    response.ReasonPhrase = String.Join(" ", tokens, 2, tokens.Length - 2);
+                }
+            }
+            if (responseHeader.StartsWith("HTTP/1.1", StringComparison.OrdinalIgnoreCase))
```

Comment:
At this point we know it starts with "HTTP/", so we should be able to do the comparison starting at index HttpStartlLength where that's defined as a constant to be 5 ("HTTP/".Length).